src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorDiagnosticFactory.cs

```diff
@@ -212,5 +212,27 @@ public static RazorDiagnostic Create_ChildContentMixedWithExplicitChildContent(S
         {
             return RazorDiagnostic.Create(ChildContentMixedWithExplicitChildContent, source ?? SourceSpan.Undefined);
         }
+
+        public static readonly RazorDiagnosticDescriptor ChildContentHasInvalidAttribute =
+            new RazorDiagnosticDescriptor(
+                "BL9997",
+                () => "Unrecognized attribute '{0}' on child content element '{1}'.",
+                RazorDiagnosticSeverity.Error);
+
+        public static RazorDiagnostic Create_ChildContentHasInvalidAttribute(SourceSpan? source, string attribute, string element)
+        {
+            return RazorDiagnostic.Create(ChildContentHasInvalidAttribute, source ?? SourceSpan.Undefined, attribute, element);
+        }
+
+        public static readonly RazorDiagnosticDescriptor ChildContentHasInvalidParameter =
+            new RazorDiagnosticDescriptor(
+                "BL9998",
+                () => "Invalid parameter name. The parameter name attribute '{0}' on child content element '{1}' can only include literal text.",
```

Comment:
Glad you anticipated this sort of thing.

Reply:
This is also kinda a cheat. The parameter name goes into an attribute, but it also gets mapped into the generated code in a way that doesn't really have a precedent in Razor. 

For instance if you put in a name that wasn't a valid c# identifier, the experience might not be  great. There's a more robust solution to this that I didn't pursue in this PR.